File: leetcode-submissions/2418_Sort_the_People.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        map<int,string>mp;
        for(int i=0;i<names.size();i++){
            mp[heights[i]]=names[i];
        }
        vector<string>ans;

        for(auto ele:mp){
            ans.push_back(ele.second);


        }
        reverse(ans.begin(),ans.end());
        return ans;
    }
};

File: leetcode-submissions/2418_Sort_the_People.ts
function sortPeople(names: string[], heights: number[]): string[] {
  // we need to create 
  const peoples=names.map((names,idx)=>({names,heights:heights[idx]}));
  peoples.sort((a,b)=>b.heights-a.heights);
  return peoples.map((p)=>p.names);
};